grants_tagger/bertmesh/model.py

```diff
@@ -0,0 +1,45 @@
+from transformers import AutoModel
+import torch
+
+
+class MultiLabelAttention(torch.nn.Module):
+    def __init__(self, D_in, num_labels):
+        super().__init__()
+        self.A = torch.nn.Parameter(torch.empty(D_in, num_labels))
+        torch.nn.init.uniform(self.A, -0.1, 0.1)
+
+    def forward(self, x):
+        attention_weights = torch.nn.functional.softmax(
+            torch.tanh(torch.matmul(x, self.A)), dim=1
+        )
+        return torch.matmul(torch.transpose(attention_weights, 2, 1), x)
+
+
+class BertMesh(torch.nn.Module):
+    def __init__(self, pretrained_model, num_labels, multilabel_attention=False):
+        super().__init__()
+        self.pretrained_model = pretrained_model
+        self.num_labels = num_labels
+        self.multilabel_attention = multilabel_attention
```

Comment:
is this line needed if you overwrite it in 2 lines below?

Reply:
🐛